tests/lib/rules/no-unsafe-optional-chaining.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * @fileoverview Tests for no-unsafe-optional-chaining rule.
+ * @author Yeon JuAn
+ */
+
+"use strict";
+
+const rule = require("../../../lib/rules/no-unsafe-optional-chaining");
+
+const { RuleTester } = require("../../../lib/rule-tester");
+
+const parserOptions = {
+    ecmaVersion: 2021,
+    sourceType: "module"
+};
+
+const ruleTester = new RuleTester({ parserOptions });
+
+ruleTester.run("no-unsafe-optional-chaining", rule, {
+    valid: [
+        "var foo;",
+        "class Foo {}",
+        "!!obj?.foo",
+        "obj?.foo();",
+        "obj?.foo?.();",
+        "(obj?.foo ?? bar)();",
+        "(obj?.foo)?.()",
+        "(obj?.foo ?? bar?.baz)?.()",
+        "(obj.foo)?.();",
+        "obj?.foo.bar;",
+        "obj?.foo?.bar;",
+        "(obj?.foo)?.bar;",
+        "(obj?.foo)?.bar.baz;",
+        "(obj?.foo)?.().bar",
+        "(obj?.foo ?? bar).baz;",
+        "(obj?.foo ?? val)`template`",
+        "new (obj?.foo ?? val)()",
+        "new bar();",
+        "obj?.foo?.()();",
+        "const {foo} = obj?.baz || {};",
+        "bar(...obj?.foo ?? []);",
+        "var bar = {...foo?.bar};",
+        "foo?.bar in {};",
+        "[foo = obj?.bar] = [];",
+        "(foo?.bar, bar)();",
+        "(foo?.bar ? baz : qux)();",
+
+        // logical operations
+        "(obj?.foo ?? bar?.baz ?? qux)();",
+        "((obj?.foo ?? bar?.baz) || qux)();",
+        "((obj?.foo || bar?.baz) || qux)();",
+        "((obj?.foo && bar?.baz) || qux)();",
+
+        // The default value option disallowArithmeticOperators is false
```

Comment:
It would be good to have a test with an explicit `disallowArithmeticOperators: false`, and one test with an empty options object (to test default value in schema):

```js
{
    code: "obj?.foo - bar;",
    options: [{}]
},
{
    code: "obj?.foo - bar;",
    options: [{
        disallowArithmeticOperators: false
    }]
},
```